[찐브로스] 2주차 우테코북 기능 구현

andole98 opened the pull request:
- 유저
* 친구 기능
* 유저 페이지
* 유저의 개인 포스트 페이지

- 게시물
* 이미지 업로드 기능
* 좋아요 기능

- 프로젝트
* 설정 프로필 구분 (local, test, deploy)

감사합니다.

andole98 committed:
feat: 2주차 기능 구현 (#122)

* refactor(service): LoginSessionManager 추가 

* refactor(user) : LoginSessionManager로 UserSession 관리

* refactor(session) : UserSession과 LoginUserDto 분리

* refactor(user) : UserService.class에 로그인한 유저를 조회하는 기능 구현

* test(user) : UserService.class 테스트 추가

* logback과 logback access 설정 파일 업로드 (#96)

* feat(logback): logback과 logback access 설정 파일들을 업로드합니다

* fix(logback): log level 설정 적용 대상 패키지명 변경

* setting(ignore): .gitignore에서 gralde-wrapper.jar 제외 (#99)

* setting(yml): profile을 local, deploy, test로 분리 (#100)

* feat(yml): application.yml 파일의 profile을 local, deploy, test로 분리

1. test환경, local에서 개발 환경, 배포시 환경을 각각 구분해주기 위해 application.yml 파일을 수정하였습니다.
2. local 환경과 배포 환경에서 mysql을 사용하도록 수정하였습니다.

* test: 사용하지 않는 테스트 클래스 제거

* 유저별 페이지 구현 (#101)

* setting(log) : .gitignore에서 log파일 제외

* feat(service) : 유저가 작성한 게시글의 목록을 조회하는 기능

* feat(controller) : UserPageController 백엔드 구현

* feat(user) : user 개인 페이지 구현

* refactor : 불필요한 코드 삭제

* refactor : UserPageController -> PostPageContorller

* hotfix : 실패 테스트 수정 (#107)

* [게시글] 게시글 수정 모달 추가 (#108)

* setting: 프로젝트 세팅

* Merge remote-tracking branch 'upstream/feat' into feat

# Conflicts:
#	.gitignore

* style(gradle): 공백 제거

* feat(front): 게시글 수정 모달 추가

* feat(front): user-page 메인 페이지 추가

* 친구 기능 백엔드 구현 (#104)

* feat(service) : 유저가 작성한 게시글의 목록을 조회하는 기능

* feat(controller) : UserPageController 백엔드 구현

* refactor : UserPageController -> PostPageContorller

* feat(repository) : 친구 기능 Repository Layer에서 구현

* feat(service) : UserService.class에 친구 추가, 조회 기능 구현

* refactor : 피드백 반영

* refactor : MYSQL에서 동작하도록 수정

* test : Email 중복 가입 테스트 추가

* setting(yml): databse-platform MySQL5Dialect -> MySQL5InnoDBDialect (#112)

* setting(yml): databse-platform 설정을 MySQL5Dialect -> MySQL5InnoDBDialect로 변경

* refactor(user): User의 join table renaming

* hotifx : existByEmail로 이메일 중복 검사 (#113)

* hotfix : 중복 가입 문제 해결 (#114)

* PostPageController에서 user-page class path 수정 (#115)

* chore(logging): 로그 설정 변경 (#116)

* feat(friend) : 친구 등록을 프론트 페이지와 연동 (#117)

* feat(friend) : 친구 등록을 프론트 페이지와 연동

* style : layout\user-page-cover -> element\user-page-cover

* test : FriendController 테스트 추가

* refactor : 프론트 엔드에서 친구 목록 출력 개선

* refactor : 유저 페이지에서 유저의 이름 표시

* Update src/main/resources/static/js/friend-add.js

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* Update src/main/resources/static/js/friend-add.js

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* Update src/main/resources/templates/element/user-tab-contents.html

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* feat: 게시글 좋아요 기능 추가 (#119)

* setting: 프로젝트 세팅

* Merge remote-tracking branch 'upstream/feat' into feat

# Conflicts:
#	.gitignore

* style(gradle): 공백 제거

* feat(front): 게시글 수정 모달 추가

* feat(domain): 게시글 도메인에 좋아요 기능 추가

* feat(service): 게시글 서비스에 좋아요 기능 추가

* feat(controller): 게시글 컨트롤러에 좋아요 기능 맵핑

* refactor(postLike): PostLike 엔티티 추가

* feat(front): 게시글 좋아요 기능 프론트와 연동

* test(postLike): 게시글 좋아요 기능 테스트 추가

* refactor(front): 좋아요 이미지와 글자를 클릭해도 좋아요 기능 작동하게 수정

* feat(mediafile): 이미지 업로드 기능 추가 (#121)

* setting: 프로젝트 세팅

* setting: 의존성 추가

1. JUnit5 추가
2. mockito 추가
3. application.yml 추가

* feat(domain): Post 생성

1. 게시글 담당 엔터티

* feat(domain): 유저 목 객체 생성

1. 임시로 사용할 유저 객체

* feat(domain): PostException 생성

1. Post 관련 예외 담당.
2. UnAuthorizedException

* test(domain): Post 테스트

1. 생성 테스트
2. 수정 테스트

* feat(domain): PostRepository 생성

1. JpaRepository

* feat(service): PostService 생성

1. CRUD 기능 제공.

* feat(service): DataAssembler 추가

1. 사용될지 모름.

* feat(user): 유저 기능 추가

* feat: User domain 객체 추가

기본적인 유효성 검사를 추가하였습니다

* feat : User.class update 구현

* feat : UserService.class 유저 생성 기능

* common: 회원 게시글 통합

1. 패키지 구조는 모듈별로 나눈다.

* [회원 기능 구현] UserService 클래스 수정, 조회, 삭제 추가했습니다 (#13)

* feat(service): UserService 수정, 삭제, 조회 구현

* refactor(domain): 이름, 비밀번호 유효성 검사 조건 변경

* feat(web): 컨트롤러, 뷰 추가 (#26)

* feat(controller): PostController 추가

* refactor(exception): 예외처리 클래스 패키지 분리

* refactor(domain): 회원 import 변경

* refactor(service): service의 메서드가 엔티티를 반환하게 수정

* test(domain): 하드코딩된 값을 상수로 추가

* test(service): 게시글 서비스 계층 테스트 추가

* feat(controller): 게시글 컨트롤러 기능 추가

* feat(controller): UserController 추가 (#22)

* refactor(package) : 테스트 패키지 분리

* feat(controller) : UserController CRUD 구현

* refactor(test) : 상수로 분리

* refactor(controller) : User가 존재하지 않을 때 404에러 코드 반환

* refacotr(controller, service) : 이메일이 중복될 때 예외 처리

1. service : UserDuplicatedException() 생성
2. controller : Accepted 응답 코드 반환

* feat(markup): 마크업 정리 (#23)

* feat(markup): base 세팅

1. 제공받은 템플릿 이동

* setting(ignore): images 추가

1. 바이너리 파일 제외

* feat(templates): 마크업 템플릿 생성

TODO:
1. 변수할당
2. 네이밍

* feat(templates): Cdn script 모듈 추가

1. common/cdns.html

* feat(front): 로그인 폼 샘플 (#29)

* setting(ignore): .gitignore 수정 (#30)

* style(main): 인덴트를 tab에서 4 space로 변경

코드 컨벤션을 위한 스타일 변경입니다.
ZenHub 사용법 등을 익히기 위해 커밋합니다.

* feat(front): 샘플로 제공된 Wootecobook 프론트엔드 코드 추가

https://woowacourse.github.io/miniprojects/wootecobook/

* chore(front): 외부 자바스크립트 라이브러리 CDN 통일

* chore(front): 외부 CSS CDN 통일 및 정리

* feat(front): 회원가입 HTML 페이지 작성

CSS 때문에 계속 시간을 잡아먹고 있는데, 일단 형상은 있으니까 커밋한다.

* edit gitignore

* gitignore 파일 수정

* feat(front): 로그인 폼 샘플

* edit .gitignore

* edit .gitignore

* delete IDE setting file

* chore(front): 바이너리 파일(이미지)은 git에서 제외

* chore(front): 중복 코드 제외

* Feat(front): 회원가입 페이지를 템플릿 기반으로 변경 (#32)

* refactor(front): signup 페이지를 Thymeleaf 템플릿 기반으로 변경

테스트해보고 싶은데 아직 라우터가 작성되지 않았다.

* refactor(front): 이미지 파일 주소를 모두 외부(github)의 것으로 하드코딩

우리 저장소에는 어떤 바이너리 파일도 집어넣지 않게 하기 위해서이다.

* feat(front): 링크 수정

웹 서버가 기본으로 돌아가고 있다는 가정 하에 메인 페이지로 가는 링크를 수정.

* feat(front): 회원가입시 입력글자수 검증기능 추가 (#36)

* feat(front): 회원가입 기능 프론트엔드 검증 추가

비밀번호 규칙은 결정되면 추가 가능.

* [유저 관련 기능] 로그인 구현 및 수정, 삭제 권한 체크 (#35)

* feat(config) : ArgumentResolver 추가

* refactor(controller, service) : 수정, 삭제할 때 권한 체크

* feat(advice) : UserControllerExceptionAdvice 추가

* refactor(repository) : UserRepository.existsByEmail 삭제

* feat(front): 로그인 페이지 템플릿 구현 (#38)

* feat(user) :  뷰에서 회원 가입 가능 (#45)

* feat(config) : ArgumentResolver 추가

* refactor(controller, service) : 수정, 삭제할 때 권한 체크

* feat(advice) : UserControllerExceptionAdvice 추가

* refactor(repository) : UserRepository.existsByEmail 삭제

* refactor(controller) : post 제어 수정

* refactor(service) : UserService.class modify, delete 메소드 수정

* refactor(exception) : Exception 클래스 리팩토링

* feat : 회원가입 프론트 연동

* feat(post): 뷰에서 게시글 작성, 삭제, 조회 기능 구현 (#43)

* test(domain): 실패하는 테스트 수정

* feat(service): 모든 게시글 조회기능 추가

* style(controller): 게시글 컨트롤러의 autowired 애노테이션 삭제

* feat: post에 회원 이름, 게시글 내용 적용

* feat(sql): 초기 데이터 등록

* feat: 프론트에서 게시글 작성기능 추가

* feat: 프론트에서 게시글 삭제기능 추가

* style: 초기 데이터 수정

* feat(user) : 유저 로그인, 로그아웃 기능 구현 (#49)

* feat(config) : ArgumentResolver 추가

* refactor(controller, service) : 수정, 삭제할 때 권한 체크

* feat(advice) : UserControllerExceptionAdvice 추가

* refactor(repository) : UserRepository.existsByEmail 삭제

* refactor(controller) : post 제어 수정

* refactor(service) : UserService.class modify, delete 메소드 수정

* refactor(exception) : Exception 클래스 리팩토링

* feat : 회원가입 프론트 연동

* feat(login) : 로그인 구현

* feat(logout) : 로그아웃 구현

* refactor(login) : 로그인 실패시 로그인 페이지로 이동

* hofix(login) : 로그인 오류 해결

* feat(login) : 로그아웃했을 때 회원 관련 정보 뷰에서 표시 제거

* refactor(session) : UserSession id 추가

* [게시글] 뷰에서 게시글 수정기능 추가 (#48)

* feat(front): 게시글 수정 템플릿 추가

* feat(front): 게시글 수정 메뉴 추가

* style(front): 게시글 생성 버튼 색 수정

* fix(front): 정적파일 경로 못 찾는 문제 해결

* feat(controller): 게시글 수정페이지 컨트롤러 추가

* build: shell script와 build.gradle dependency, gradle-wrapper 추가 (#53)

* feat(front): 로그인 페이지 템플릿 구현

* feat(deploy): graddle-wrapper 추가

* feat(deploy): build 하기 전에 test를 수행하기 위해 build.gradle에 필요한 dependency를 추가합니다

* feat(deploy): 자동 배포 shell script를 추가합니다

* feat(user): 회원 정보 수정, 삭제 기능 추가

* feat(user) : 이름 수정 기능

* test : UserController 테스트 추가

* refactor(exception) : 인가되지 않은 사용자가 수정페이지로 갈 때 index로 이동

* feat(user) : 유저 수정할 때 예외 발생시 에러 메시지 출력

* refactor(user) : ResponseMessage 정적 팩토리 메소드 추가

* style(edit) : 유저 수정 페이지에 headline 추가

* test(login, logout) : 로그인, 로그아웃 테스트 클래스 추가

* refactor(login) : 로그인 성공시 "/" 로 리다이렉트

* feat(delete) : 유저 탈퇴 기능 구현 (프론트엔드, 백엔드)

* refactor : 컨벤션 리팩토링

* style(user) : 유저 수정 페이지 개선

* refactor(exception) : Test Coverage 상승

* feat(post): 로그인 상태에 따라 게시글 작성, 수정, 삭제하도록 수정 (#64)

* refactor(controller): ArgumentResolver를 통해 로그인한 회원을 주입하여 게시글 수정하도록 변경

* refactor(controller): ArgumentResolver를 통해 로그인한 회원을 주입하여 게시글 작성하도록 변경

* refactor(controller): ArgumentResolver를 통해 로그인한 회원을 주입하여 게시글 삭제하도록 변경

* refactor: PostService가 UserService를 참조하도록 수정

* feat: 게시글과 관련된 예외처리

* refactor(post): PostService에서 UserSession 참조 삭제

* 세션 대신 Id 전달

* refactor(package): 패키지 구조 통일

* fix: login.html 폼 태그 변경

* UserRequestDto 매핑

* fix: deploy.sh headerline 추가

* setting(ignore): gradle ignore

* refactor(exception): 예외 체인 변경

* refactor(exception) : 이메일 중복 있을 때 예외 연쇄

* refactor(post) : PostRequestDto에서 Post 생성

* refactor(js): 마크업, js 분리 (#73)

* refactor(markup): 마크업 정리

* js, html 분리
* common -> layout 폴더명 변경
* layout, element 구분

** layout은 레이아웃을 담당
** element는 레이아웃을 채우는 요소들

* refactor(js): fetch Api 템플릿 생성

* 각 모듈 js 분리

* refactor(js): 이벤트 등록 리팩토링

* 익명함수 콜

* fix(js): post-edit.js 버그 해결

* addEventListner 위치 조정
* common 디렉토리 추가

* fix(test): 테스트 수정

* UserController put 방식을 변경
* json 방식 -> form 방식

* fix(templates): thymeleaf 경로 수정 (#79)

* thymeleaf는 절대경로가 없네요.

* refactor(common): AppConfig 패키지 위치 변경

* user -> common

* feat(domain): MediaFile 엔터티 생성

* MediaFile은 필요한 엔터티가 가져가 쓸 수 있음. OneToMany

* feat(web): Upload 파일 서포트 패키지 생성

1. 어노테이션 UploadedFile
2. MultipartFile 래퍼 클래스 UploadFile
3. UploadFile Resolver

* feat(controller): PostWithImageController 생성

* 이미지와 함께 등록한 Post 처리용

* feat(markup): 이미지 등록 모달, 뷰 생성

* feat(js): social-feed.js 생성

* 피드 작성 인터랙션

* refactor(common): AppConfig 패키지 위치 조정

user -> common

* refactor(js): social-feed.js IIFE

* refactor(support): UploadTo 인터페이스 생성

* 저장 전략에 따라 UploadToLocal, UploadToS3로 구성할 예정

* fix(conflict): 충돌 해결

andole98 committed:
Delete TestForLearn.java

wmakerjun commented on src/main/resources/static/js/friend-add.js:
```diff
@@ -0,0 +1,13 @@
+(function() {
+    const sendFriendRequest = (event)=>{
```
공백 컨벤션을 맞춰주면 좋을 것 같아요 :)
```(event) => { ```

wmakerjun commented on src/main/resources/static/js/friend-add.js:
```diff
@@ -0,0 +1,13 @@
+(function() {
+    const sendFriendRequest = (event)=>{
+        const requestFriendId = event.target.dataset.userId;
+        const api = new AjaxApi();
```
api를 이용하는 곳들에서 매번 AjaxApi 객체의 인스턴스를 만들고 있는 특별한 이유가 있을까요? 서로 다른 인스턴스를 만들어서 활용할 목적으로 AjaxApi를 쓸 것 같지는 않은데 생성자 형태로 만든 이유가 궁금합니다~! 또 fetch-api.js 파일에서 this.bodyData를 매번 인스턴스를 생성하는 것도 왜 그렇게 진행한 것인지 궁금합니다! 한번 할당된 객체 형태를 재사용하는 방식으로 이용해도 좋을 것 같아요~
[링크](https://heygyun.tistory.com/45)도 같이 참조하면 좋을 것 같습니다 :)

wmakerjun commented on src/main/resources/static/js/social-feed.js:
```diff
@@ -12,5 +12,40 @@
+        preview.file = file;
+
+        const reader = new FileReader();
+        reader.onload = (function(aImg) {
```
arrow function과 일반 function 표기법이 혼재되어 있습니다. 여기서도 다른 부분과 동일하게 arrow function으로 맞춰주면 좋을 것 같아요 :)

wmakerjun reviewed (changes requested):
코멘트 남길게 별로 없을 정도로 너무 멋지게 구현하고 있습니다! html 코드도 layout과 element별로 나누어서 잘 정리하고 있는 것 같아요!!!! 💯💯  자바스크립트도 많은 노력을 들여서 구현한 흔적이 많이 보입니다 ㅎㅎ 그래서 딱 3가지만 코멘트 남겼는데요.
1. 공백 컨벤션
2. 자바스크립트 표현방식 동일화
3. 인스턴스 생성
위 3가지에 대한 것만 다시한번 빠르게 확인해주면 좋을 것 같아요 :)

andole98 committed:
feat: 추가 기능 구현 (#140)

* feat(Comment): 코멘트 기능 관련 백엔드 코드 추가 (#123)

며칠 전 코드 기반이라 지금의 코드와는 맞지 않는 부분들이 좀 있는데,
merge된 이후 수정할 예정.

* feat(friend): 친구에 연좌제 반영 (탈퇴, 자신을 친구 추가하는 것 금지) (#128)

* refactor : 자기 자신을 친구 추가 할 수 없음

* refactor(friend) : 삼족을 멸할 수 있도록 수정

* refactor(markup): Head 태그 변경 (#131)

* fragment -> th:block

* refactor(imagefile): 업로드 전략에 따라 업로드할 수 있게 변경 (#135)

* setting(gradle): AWS SDK 추가

* feat(common): 공통 미디어 파일 핸들링

* UploadTo 인터페이스
* UploadTo 구현체 : UploadToLocal, UploadToS3
* UploadToFactory : properties 기반으로 객체 생성
* UploadFileResolver : 컴포넌트, 파라미터 리졸빙

* setting(properties): 커스텀 프로퍼티 추가

* test(mediafile): AuthedWebTestClient 변경

* 친구 기능 삭제 버그 수정 (#136)

* refactor(friend) : 친구 구조 수정 (TestUser로 수정)

- 주석 존재
- 중간 단계

* refactor : 회원 삭제할 때 버그 수정

회원 삭제할 때 친구 관계인 친구를 삭제해버리는 버그 수정

* refactor : 주석 제거

* refactor : NotFound 어노테이션 삭제

* refactor : FetchType.LAZY로 수정

* refactor : @Transactional 추가

* Update src/main/java/com/woowacourse/zzinbros/user/domain/repository/UserRepository.java

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* Update src/main/resources/templates/element/headline-user.html

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* Update src/main/resources/templates/element/right-sidebar-detail.html

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* Update src/main/resources/templates/element/left-profile-detail.html

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* Update src/main/resources/templates/layout/headline.html

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* Update src/main/resources/templates/element/right-sidebar-detail.html

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* fix(test): 테스트 프로필 수정 (#137)

* refactor(Comment): 아침의 팀내 피드백 반영 (#134)

* test(comment): BaseTest 메소드 상속

* refactor(comment): Controller에서 UserService의 기능을 사용하도록 변경

* refactor(comment): Service에서 메소드 시그니쳐의 Long을 long 프리미티브로 변경

* test(comment): 미사용 MockBean 삭제

* test(comment): 필요없는 Autowired Repository 삭제

* test(comment): 도메인 객체의 로직 메소드에 대한 테스트 코드 추가

* feat(comment): 도메인 객체 및 Service의 변경

 * Post 삭제시 해당 게시물에 달린 Comment도 함께 삭제되게끔 정책 변경
 * 댓글 삭제시 명시적으로 작성자 및 게시물과의 관계를 끊도록 변경
 * 동일성 확인을 고유 ID값으로만 하게끔 변경

* feat(comment): 컨트롤러에서 명시적으로 ResponseEntity로 응답 생성

이제 명시적으로 ResponseEntity를 통해 HTTP status code를 지정하여 응답한다.

* refactor(comment): User 쪽 변경사항 반영

andole98 committed:
refactor: 피드백 반영 및 프론트 페이지 구현 (#153)

* Delete Project_Default.xml

* chore(Log settings): 로그 관련 설정 적용 (#147)

* chore(logback): 로그 관련 설정 적용

 * 기본 `logback.xml` 대신 Spring Boot의 로그 설정을 사용하게끔 바꾸었습니다.
 * 개발환경/테스트/운영서버의 로그 수준을 모두 다르게 하였습니다.
 * 로컬에서 서비스를 띄우면 콘솔 창에 Request와 Response가 출력되게끔 하였습니다.
 * (지원되는 환경에서) 로그에 색이 입혀져서 출력됩니다.

* style(application.yml): indent를 4 space로 수정, 필요없는 빈 줄 삭제

* fix(logback-access.xml): 의도하지 않은 줄바꿈 삽입 제거 (#149)

IDE에서 자동 리포맷을 하는 바람에 의도하지 않은 줄바꿈이 로그 출력에 들어가는 문제를 해결하였습니다.

* fix(logback-access.xml): 빠진 줄바꿈 문자 추가

* fix(test): S3 reponseTimeout 변경 (#151)

- 5000 -> 10000

* refactor(js): 피드백 반영 (#148)

- AjaxApi 인스턴스를 전역으로 변경
- AjaxApi 내부 bodyData를 오브젝트 리터럴로 변경

* feat(markup): Entrance page 추가 (#145)

* refactor(service): LoginSessionManager 추가 

* refactor(user) : LoginSessionManager로 UserSession 관리

* refactor(session) : UserSession과 LoginUserDto 분리

* refactor(user) : UserService.class에 로그인한 유저를 조회하는 기능 구현

* test(user) : UserService.class 테스트 추가

* logback과 logback access 설정 파일 업로드 (#96)

* feat(logback): logback과 logback access 설정 파일들을 업로드합니다

* fix(logback): log level 설정 적용 대상 패키지명 변경

* setting(ignore): .gitignore에서 gralde-wrapper.jar 제외 (#99)

* setting(yml): profile을 local, deploy, test로 분리 (#100)

* feat(yml): application.yml 파일의 profile을 local, deploy, test로 분리

1. test환경, local에서 개발 환경, 배포시 환경을 각각 구분해주기 위해 application.yml 파일을 수정하였습니다.
2. local 환경과 배포 환경에서 mysql을 사용하도록 수정하였습니다.

* test: 사용하지 않는 테스트 클래스 제거

* 유저별 페이지 구현 (#101)

* setting(log) : .gitignore에서 log파일 제외

* feat(service) : 유저가 작성한 게시글의 목록을 조회하는 기능

* feat(controller) : UserPageController 백엔드 구현

* feat(user) : user 개인 페이지 구현

* refactor : 불필요한 코드 삭제

* refactor : UserPageController -> PostPageContorller

* hotfix : 실패 테스트 수정 (#107)

* [게시글] 게시글 수정 모달 추가 (#108)

* setting: 프로젝트 세팅

* Merge remote-tracking branch 'upstream/feat' into feat

# Conflicts:
#	.gitignore

* style(gradle): 공백 제거

* feat(front): 게시글 수정 모달 추가

* feat(front): user-page 메인 페이지 추가

* 친구 기능 백엔드 구현 (#104)

* feat(service) : 유저가 작성한 게시글의 목록을 조회하는 기능

* feat(controller) : UserPageController 백엔드 구현

* refactor : UserPageController -> PostPageContorller

* feat(repository) : 친구 기능 Repository Layer에서 구현

* feat(service) : UserService.class에 친구 추가, 조회 기능 구현

* refactor : 피드백 반영

* refactor : MYSQL에서 동작하도록 수정

* test : Email 중복 가입 테스트 추가

* setting(yml): databse-platform MySQL5Dialect -> MySQL5InnoDBDialect (#112)

* setting(yml): databse-platform 설정을 MySQL5Dialect -> MySQL5InnoDBDialect로 변경

* refactor(user): User의 join table renaming

* hotifx : existByEmail로 이메일 중복 검사 (#113)

* hotfix : 중복 가입 문제 해결 (#114)

* PostPageController에서 user-page class path 수정 (#115)

* chore(logging): 로그 설정 변경 (#116)

* feat(friend) : 친구 등록을 프론트 페이지와 연동 (#117)

* feat(friend) : 친구 등록을 프론트 페이지와 연동

* style : layout\user-page-cover -> element\user-page-cover

* test : FriendController 테스트 추가

* refactor : 프론트 엔드에서 친구 목록 출력 개선

* refactor : 유저 페이지에서 유저의 이름 표시

* Update src/main/resources/static/js/friend-add.js

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* Update src/main/resources/static/js/friend-add.js

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* Update src/main/resources/templates/element/user-tab-contents.html

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* feat: 게시글 좋아요 기능 추가 (#119)

* setting: 프로젝트 세팅

* Merge remote-tracking branch 'upstream/feat' into feat

# Conflicts:
#	.gitignore

* style(gradle): 공백 제거

* feat(front): 게시글 수정 모달 추가

* feat(domain): 게시글 도메인에 좋아요 기능 추가

* feat(service): 게시글 서비스에 좋아요 기능 추가

* feat(controller): 게시글 컨트롤러에 좋아요 기능 맵핑

* refactor(postLike): PostLike 엔티티 추가

* feat(front): 게시글 좋아요 기능 프론트와 연동

* test(postLike): 게시글 좋아요 기능 테스트 추가

* refactor(front): 좋아요 이미지와 글자를 클릭해도 좋아요 기능 작동하게 수정

* feat(mediafile): 이미지 업로드 기능 추가 (#121)

* setting: 프로젝트 세팅

* setting: 의존성 추가

1. JUnit5 추가
2. mockito 추가
3. application.yml 추가

* feat(domain): Post 생성

1. 게시글 담당 엔터티

* feat(domain): 유저 목 객체 생성

1. 임시로 사용할 유저 객체

* feat(domain): PostException 생성

1. Post 관련 예외 담당.
2. UnAuthorizedException

* test(domain): Post 테스트

1. 생성 테스트
2. 수정 테스트

* feat(domain): PostRepository 생성

1. JpaRepository

* feat(service): PostService 생성

1. CRUD 기능 제공.

* feat(service): DataAssembler 추가

1. 사용될지 모름.

* feat(user): 유저 기능 추가

* feat: User domain 객체 추가

기본적인 유효성 검사를 추가하였습니다

* feat : User.class update 구현

* feat : UserService.class 유저 생성 기능

* common: 회원 게시글 통합

1. 패키지 구조는 모듈별로 나눈다.

* [회원 기능 구현] UserService 클래스 수정, 조회, 삭제 추가했습니다 (#13)

* feat(service): UserService 수정, 삭제, 조회 구현

* refactor(domain): 이름, 비밀번호 유효성 검사 조건 변경

* feat(web): 컨트롤러, 뷰 추가 (#26)

* feat(controller): PostController 추가

* refactor(exception): 예외처리 클래스 패키지 분리

* refactor(domain): 회원 import 변경

* refactor(service): service의 메서드가 엔티티를 반환하게 수정

* test(domain): 하드코딩된 값을 상수로 추가

* test(service): 게시글 서비스 계층 테스트 추가

* feat(controller): 게시글 컨트롤러 기능 추가

* feat(controller): UserController 추가 (#22)

* refactor(package) : 테스트 패키지 분리

* feat(controller) : UserController CRUD 구현

* refactor(test) : 상수로 분리

* refactor(controller) : User가 존재하지 않을 때 404에러 코드 반환

* refacotr(controller, service) : 이메일이 중복될 때 예외 처리

1. service : UserDuplicatedException() 생성
2. controller : Accepted 응답 코드 반환

* feat(markup): 마크업 정리 (#23)

* feat(markup): base 세팅

1. 제공받은 템플릿 이동

* setting(ignore): images 추가

1. 바이너리 파일 제외

* feat(templates): 마크업 템플릿 생성

TODO:
1. 변수할당
2. 네이밍

* feat(templates): Cdn script 모듈 추가

1. common/cdns.html

* feat(front): 로그인 폼 샘플 (#29)

* setting(ignore): .gitignore 수정 (#30)

* style(main): 인덴트를 tab에서 4 space로 변경

코드 컨벤션을 위한 스타일 변경입니다.
ZenHub 사용법 등을 익히기 위해 커밋합니다.

* feat(front): 샘플로 제공된 Wootecobook 프론트엔드 코드 추가

https://woowacourse.github.io/miniprojects/wootecobook/

* chore(front): 외부 자바스크립트 라이브러리 CDN 통일

* chore(front): 외부 CSS CDN 통일 및 정리

* feat(front): 회원가입 HTML 페이지 작성

CSS 때문에 계속 시간을 잡아먹고 있는데, 일단 형상은 있으니까 커밋한다.

* edit gitignore

* gitignore 파일 수정

* feat(front): 로그인 폼 샘플

* edit .gitignore

* edit .gitignore

* delete IDE setting file

* chore(front): 바이너리 파일(이미지)은 git에서 제외

* chore(front): 중복 코드 제외

* Feat(front): 회원가입 페이지를 템플릿 기반으로 변경 (#32)

* refactor(front): signup 페이지를 Thymeleaf 템플릿 기반으로 변경

테스트해보고 싶은데 아직 라우터가 작성되지 않았다.

* refactor(front): 이미지 파일 주소를 모두 외부(github)의 것으로 하드코딩

우리 저장소에는 어떤 바이너리 파일도 집어넣지 않게 하기 위해서이다.

* feat(front): 링크 수정

웹 서버가 기본으로 돌아가고 있다는 가정 하에 메인 페이지로 가는 링크를 수정.

* feat(front): 회원가입시 입력글자수 검증기능 추가 (#36)

* feat(front): 회원가입 기능 프론트엔드 검증 추가

비밀번호 규칙은 결정되면 추가 가능.

* [유저 관련 기능] 로그인 구현 및 수정, 삭제 권한 체크 (#35)

* feat(config) : ArgumentResolver 추가

* refactor(controller, service) : 수정, 삭제할 때 권한 체크

* feat(advice) : UserControllerExceptionAdvice 추가

* refactor(repository) : UserRepository.existsByEmail 삭제

* feat(front): 로그인 페이지 템플릿 구현 (#38)

* feat(user) :  뷰에서 회원 가입 가능 (#45)

* feat(config) : ArgumentResolver 추가

* refactor(controller, service) : 수정, 삭제할 때 권한 체크

* feat(advice) : UserControllerExceptionAdvice 추가

* refactor(repository) : UserRepository.existsByEmail 삭제

* refactor(controller) : post 제어 수정

* refactor(service) : UserService.class modify, delete 메소드 수정

* refactor(exception) : Exception 클래스 리팩토링

* feat : 회원가입 프론트 연동

* feat(post): 뷰에서 게시글 작성, 삭제, 조회 기능 구현 (#43)

* test(domain): 실패하는 테스트 수정

* feat(service): 모든 게시글 조회기능 추가

* style(controller): 게시글 컨트롤러의 autowired 애노테이션 삭제

* feat: post에 회원 이름, 게시글 내용 적용

* feat(sql): 초기 데이터 등록

* feat: 프론트에서 게시글 작성기능 추가

* feat: 프론트에서 게시글 삭제기능 추가

* style: 초기 데이터 수정

* feat(user) : 유저 로그인, 로그아웃 기능 구현 (#49)

* feat(config) : ArgumentResolver 추가

* refactor(controller, service) : 수정, 삭제할 때 권한 체크

* feat(advice) : UserControllerExceptionAdvice 추가

* refactor(repository) : UserRepository.existsByEmail 삭제

* refactor(controller) : post 제어 수정

* refactor(service) : UserService.class modify, delete 메소드 수정

* refactor(exception) : Exception 클래스 리팩토링

* feat : 회원가입 프론트 연동

* feat(login) : 로그인 구현

* feat(logout) : 로그아웃 구현

* refactor(login) : 로그인 실패시 로그인 페이지로 이동

* hofix(login) : 로그인 오류 해결

* feat(login) : 로그아웃했을 때 회원 관련 정보 뷰에서 표시 제거

* refactor(session) : UserSession id 추가

* [게시글] 뷰에서 게시글 수정기능 추가 (#48)

* feat(front): 게시글 수정 템플릿 추가

* feat(front): 게시글 수정 메뉴 추가

* style(front): 게시글 생성 버튼 색 수정

* fix(front): 정적파일 경로 못 찾는 문제 해결

* feat(controller): 게시글 수정페이지 컨트롤러 추가

* build: shell script와 build.gradle dependency, gradle-wrapper 추가 (#53)

* feat(front): 로그인 페이지 템플릿 구현

* feat(deploy): graddle-wrapper 추가

* feat(deploy): build 하기 전에 test를 수행하기 위해 build.gradle에 필요한 dependency를 추가합니다

* feat(deploy): 자동 배포 shell script를 추가합니다

* feat(user): 회원 정보 수정, 삭제 기능 추가

* feat(user) : 이름 수정 기능

* test : UserController 테스트 추가

* refactor(exception) : 인가되지 않은 사용자가 수정페이지로 갈 때 index로 이동

* feat(user) : 유저 수정할 때 예외 발생시 에러 메시지 출력

* refactor(user) : ResponseMessage 정적 팩토리 메소드 추가

* style(edit) : 유저 수정 페이지에 headline 추가

* test(login, logout) : 로그인, 로그아웃 테스트 클래스 추가

* refactor(login) : 로그인 성공시 "/" 로 리다이렉트

* feat(delete) : 유저 탈퇴 기능 구현 (프론트엔드, 백엔드)

* refactor : 컨벤션 리팩토링

* style(user) : 유저 수정 페이지 개선

* refactor(exception) : Test Coverage 상승

* feat(post): 로그인 상태에 따라 게시글 작성, 수정, 삭제하도록 수정 (#64)

* refactor(controller): ArgumentResolver를 통해 로그인한 회원을 주입하여 게시글 수정하도록 변경

* refactor(controller): ArgumentResolver를 통해 로그인한 회원을 주입하여 게시글 작성하도록 변경

* refactor(controller): ArgumentResolver를 통해 로그인한 회원을 주입하여 게시글 삭제하도록 변경

* refactor: PostService가 UserService를 참조하도록 수정

* feat: 게시글과 관련된 예외처리

* refactor(post): PostService에서 UserSession 참조 삭제

* 세션 대신 Id 전달

* refactor(package): 패키지 구조 통일

* fix: login.html 폼 태그 변경

* UserRequestDto 매핑

* fix: deploy.sh headerline 추가

* setting(ignore): gradle ignore

* refactor(exception): 예외 체인 변경

* refactor(exception) : 이메일 중복 있을 때 예외 연쇄

* refactor(post) : PostRequestDto에서 Post 생성

* refactor(js): 마크업, js 분리 (#73)

* refactor(markup): 마크업 정리

* js, html 분리
* common -> layout 폴더명 변경
* layout, element 구분

** layout은 레이아웃을 담당
** element는 레이아웃을 채우는 요소들

* refactor(js): fetch Api 템플릿 생성

* 각 모듈 js 분리

* refactor(js): 이벤트 등록 리팩토링

* 익명함수 콜

* fix(js): post-edit.js 버그 해결

* addEventListner 위치 조정
* common 디렉토리 추가

* fix(test): 테스트 수정

* UserController put 방식을 변경
* json 방식 -> form 방식

* fix(templates): thymeleaf 경로 수정 (#79)

* thymeleaf는 절대경로가 없네요.

* refactor(common): AppConfig 패키지 위치 변경

* user -> common

* feat(domain): MediaFile 엔터티 생성

* MediaFile은 필요한 엔터티가 가져가 쓸 수 있음. OneToMany

* feat(web): Upload 파일 서포트 패키지 생성

1. 어노테이션 UploadedFile
2. MultipartFile 래퍼 클래스 UploadFile
3. UploadFile Resolver

* feat(controller): PostWithImageController 생성

* 이미지와 함께 등록한 Post 처리용

* feat(markup): 이미지 등록 모달, 뷰 생성

* feat(js): social-feed.js 생성

* 피드 작성 인터랙션

* refactor(common): AppConfig 패키지 위치 조정

user -> common

* refactor(js): social-feed.js IIFE

* refactor(support): UploadTo 인터페이스 생성

* 저장 전략에 따라 UploadToLocal, UploadToS3로 구성할 예정

* fix(conflict): 충돌 해결

* fix(test): 학습테스트 제거 (#126)

* feat(Comment): 코멘트 기능 관련 백엔드 코드 추가 (#123)

며칠 전 코드 기반이라 지금의 코드와는 맞지 않는 부분들이 좀 있는데,
merge된 이후 수정할 예정.

* feat(friend): 친구에 연좌제 반영 (탈퇴, 자신을 친구 추가하는 것 금지) (#128)

* refactor : 자기 자신을 친구 추가 할 수 없음

* refactor(friend) : 삼족을 멸할 수 있도록 수정

* refactor(markup): Head 태그 변경 (#131)

* fragment -> th:block

* refactor(imagefile): 업로드 전략에 따라 업로드할 수 있게 변경 (#135)

* setting(gradle): AWS SDK 추가

* feat(common): 공통 미디어 파일 핸들링

* UploadTo 인터페이스
* UploadTo 구현체 : UploadToLocal, UploadToS3
* UploadToFactory : properties 기반으로 객체 생성
* UploadFileResolver : 컴포넌트, 파라미터 리졸빙

* setting(properties): 커스텀 프로퍼티 추가

* test(mediafile): AuthedWebTestClient 변경

* 친구 기능 삭제 버그 수정 (#136)

* refactor(friend) : 친구 구조 수정 (TestUser로 수정)

- 주석 존재
- 중간 단계

* refactor : 회원 삭제할 때 버그 수정

회원 삭제할 때 친구 관계인 친구를 삭제해버리는 버그 수정

* refactor : 주석 제거

* refactor : NotFound 어노테이션 삭제

* refactor : FetchType.LAZY로 수정

* refactor : @Transactional 추가

* Update src/main/java/com/woowacourse/zzinbros/user/domain/repository/UserRepository.java

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* Update src/main/resources/templates/element/headline-user.html

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* Update src/main/resources/templates/element/right-sidebar-detail.html

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* Update src/main/resources/templates/element/left-profile-detail.html

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* Update src/main/resources/templates/layout/headline.html

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* Update src/main/resources/templates/element/right-sidebar-detail.html

Co-Authored-By: JYK <40727649+andole87@users.noreply.github.com>

* fix(test): 테스트 프로필 수정 (#137)

* refactor(Comment): 아침의 팀내 피드백 반영 (#134)

* test(comment): BaseTest 메소드 상속

* refactor(comment): Controller에서 UserService의 기능을 사용하도록 변경

* refactor(comment): Service에서 메소드 시그니쳐의 Long을 long 프리미티브로 변경

* test(comment): 미사용 MockBean 삭제

* test(comment): 필요없는 Autowired Repository 삭제

* test(comment): 도메인 객체의 로직 메소드에 대한 테스트 코드 추가

* feat(comment): 도메인 객체 및 Service의 변경

 * Post 삭제시 해당 게시물에 달린 Comment도 함께 삭제되게끔 정책 변경
 * 댓글 삭제시 명시적으로 작성자 및 게시물과의 관계를 끊도록 변경
 * 동일성 확인을 고유 ID값으로만 하게끔 변경

* feat(comment): 컨트롤러에서 명시적으로 ResponseEntity로 응답 생성

이제 명시적으로 ResponseEntity를 통해 HTTP status code를 지정하여 응답한다.

* refactor(comment): User 쪽 변경사항 반영

* feat(front): userpage 관련 html 템플릿 디렉토리 구조 변경

layout과 element 디렉토리 내에 각각 userpage 디렉토리를 만들어 관리하도록 수정합니다.

* feat(css): header 테마 색깔 변경 및 테마 색 버튼 추가

#303030을 tecobook 메인 테마 색으로 정하였습니다.

* feat(front): entrance 페이지 추가

1. 페이스북 로그인 페이지와 마찬가지로 로그인 및 회원가입을 함께 처리할 수 있도록
entrance 페이지를 추가하였습니다.

* feat(front): headline에서 불필요한 정보 제거

1. 검색 기능을 제거하였습니다.
2. index 페이지에서 보이는 headline에서 더이상 로그인과 회원 가입 기능을 연결해줄 필요가
 없어져 제거하였습니다. 차후 정리해야할 내용은 다음과 같습니다.
 - 기존 login페이지와 signup 페이지를 삭제하고, 백엔드에서 컨트롤러를 수정해야 합니다.
 - 로그인하지 않고 index페이지로 접근하는 경우, index페이지에 접근하지 못하도록 수정해야 합니다.
entrance 페이지로 redirect 해주는 방법 등을 고려해보도록 합니다.

* feat(front): social feed의 게시하기 버튼 수정

색상과 크기, 이모티콘을 수정하였습니다.

* fix(entrance): entrance 머지 과정중 버그 해결

1. entrance header에 수정사항 반영
2. LoginControllerTest의 loginFail 메서드를 LoginController의 수정사항에
맞춰 실패하는 테스트 수정

* feat: domain 객체간 중복 제거를 위해 BaseEntity를 추가 (#150)

* fix(test): S3 reponseTimeout 변경 (#152)

- 5000 -> 10000

andole98 committed:
setting(images): 기본 이미지 git에 추가 (#156)

* test: 게시글 컨트롤러 테스트 추가 (#154)

* setting: 프로젝트 세팅

* Merge remote-tracking branch 'upstream/feat' into feat

# Conflicts:
#	.gitignore

* test(controller): 게시글 컨트롤러 테스트 추가

* fix(gitignore): git ignore가 .idea 폴더를 무시하는 문제 해결

* feat(images): 기본 이미지들을 추가합니다 (#155)

Deocksoo committed:
fix: login 요청을 entrance 페이지로 리다이렉트 (#158)

* test: 게시글 컨트롤러 테스트 추가 (#154)

* setting: 프로젝트 세팅

* Merge remote-tracking branch 'upstream/feat' into feat

# Conflicts:
#	.gitignore

* test(controller): 게시글 컨트롤러 테스트 추가

* fix(gitignore): git ignore가 .idea 폴더를 무시하는 문제 해결

* feat(images): 기본 이미지들을 추가합니다 (#155)

* fix: login 요청이 들어오는 경우 entrance 페이지로 redirect (#157)

* feat(images): 기본 이미지들을 추가합니다

* fix: 메인 페이지 로고 사이즈 조절

* fix: login 요청을 하는 경우 entrance로 리다이렉트 하도록 합니다
